duneanaobj/StandardRecord/SRInteractionBranch.h

```diff
@@ -21,7 +21,8 @@ namespace caf
       std::vector<SRInteraction> pandora;   ///< Interactions from Pandora reconstruction
       std::size_t npandora;
 
-
+      bool isselected = true; ///< Is this interaction selected by the analysis?
```

Comment:
Do you really (intend to) do the whole selection process upstream of CAF-making?  Most CAF users that I'm aware of take the CAFs as _input_, and make their selection decisions based on stuff in the CAF itself.  At that point the CAF is read-only, and a flag like this is not usable because it's not settable.  Is your workflow different than that?

Reply:
So, I guess that this is something to be discussed in general. As you pointed out this flag is not meant to be used during the generation but rather later. Indeed, MaCh3 for now makes use of the CAFs directly as input and there is then no way to add extra information beyond what is in the CAFs. For the atmospherics we have some pre-processing stage of the CAFs that split the samples into different channels and adds the flux weight information using the the `genweight` field in the CAF format. Thta is at this stage that I would like to have such a flag to tag if an event ise selected or not to pass the info to MaCh3 without having to actually discard some events.

Reply:
Just noticed I didn't follow up on this thread.  Thanks for explaining.  I think perhaps the way to create the least ongoing confusion is to call this field `preselected`, with a Doxygen comment identifying the expected use case:
```suggestion
      bool preselected = false; ///< Useful for workflows where CAFs are preprocessed / filtered in the process of making concatenated files
```

Reply:
Applied